unimod_mapper/unimod_mapper.py

```diff
@@ -688,6 +719,165 @@ def _reparseXML(self, xml_file_list=[]):
         self._data_list = self._parseXML(xml_file_list=xml_file_list)
         self._mapper = self._initialize_mapper()
 
+    def map_mods(self, mod_list):
+        """
+        Maps modifications defined in params["modification"] using unimod. Using the
+        dict format for the mods, the dict can be adjusted depending on the purpose of
+        the mapping. Moreover, it can only have the minimal amound of items, or e.g.
+        engine-specific ones. At the end the mapped vallues will be updated to the
+        original dict.
+
+        Args:
+            mod_list (list): list of mod_dicts containing all relevant info about a
+                             given modification
+
+        Returns:
+            rdict (dict): dict with mod types as keys and corresponding lists of
+                             mod dicts mapped to unimod
+
+        Examples:
+
+            mod_list = [
+                {
+                    "aa": "M",              # specify the modified amino acid as a single letter, use '*' if the amino acid is variable
+                    "type": "opt",          # specify if it is a fixed (fix) or potential (opt) modification
+                    "position": "any",      # specify the position within the protein/peptide (Prot-N-term, Prot-C-term), use 'any' if the positon is variable
+                    "name": "Oxidation",    # specify the unimod PSI-MS Name (alternative to id)
+                    "id": None,             # specify the unimod Accession (alternative to name)
+                    "composition": None,    # For user-defined mods composition needs to be given as a Hill notation
+                }
+            ]
+
+        """
+
+        rdict = {"fix": [], "opt": []}
+        for index, mod in enumerate(mod_list):
+
+            # Check if any key from the mod dict is not valid!
+            allowed_keys = ["aa", "type", "position", "name", "id", "composition",
+                            "max_num_per_peptide", "intern_dist", "required", "neutral_loss"]
+            wrong_keys = []
+            for key in mod.keys():
+                if key not in allowed_keys:
+                    wrong_keys.append(key)
+            if wrong_keys != []:
```

Comment:
Not really necessary to collect all the wrong keys here (since nothing is done with them afterwards).
You can log the warning and break the loop after the first wrong key.

Reply:
true, removed the unnecessary collection ;)